snap/snapcraft.yaml

```diff
@@ -10,14 +10,26 @@ description: |
 grade: stable
 confinement: strict
 
+system-usernames:
+  snap_daemon: shared
+
+hooks:
+  install:
+    plugs:
+      - network
+      - network-bind
+
 apps:
   init:
     command: with_config.sh bin/zkServer-initialize.sh
   cli:
     command: with_config.sh bin/zkCli.sh
   daemon:
     command: with_config.sh bin/zkServer.sh start
-    plugs: [network, network-bind, removable-media]
+    plugs:
+      - network
+      - network-bind
+      - removable-media
```

Comment:
You could request this interface to be auto-connect on the [forum](forum.snapcraft.io/), however [from the docs](https://snapcraft.io/docs/removable-media-interface) it doesn't seem like it's often granted:
> This interface is typically [not granted](https://snapcraft.io/docs/process-for-aliases-auto-connections-and-tracks) the ability to [auto-connect](https://snapcraft.io/docs/interface-management#heading--auto-connections)...

I'm not sure where you need it, but if it's crucial to your snap, you'll have to do a safe check where you need it (i.e: daemon start) - in case you are not granted auto-connect on this interface: 
```
if ! snapctl is-connected "removable-media";
then
    echo "Please run the following command: sudo snap connect zookeeper:removable-media"
    echo "Then run: sudo snap start zookeeper.daemon"
    exit 1
fi
```

Reply:
Agreed, it was hard to get, so I added this to the `operator-libs-linux/snap.py` - https://github.com/canonical/operator-libs-linux/blob/e22cd16b648418b4d06d2b3813f36d9f9c5534da/lib/charms/operator_libs_linux/v1/snap.py#L371-L395 so that we could handle it from the charm side.

Reply:
I think that the safeguard must happen on the snap side, for users who might only be interested in the snap itself prior to using it - so that they don't start a server instance that might fail later, no?